src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -3694,54 +3705,58 @@ string WriteProperty(byte[] propertyName, byte[] value)
             Assert.Equal(expected3, result);
         }
 
+        private const string InvalidUtf16 = "\"\\uFFFDa\"";
+        private const string ValidUtf16 = "\"\\uD801\\uDC37\"";
+
         [Theory]
-        [InlineData(true, true)]
-        [InlineData(true, false)]
-        [InlineData(false, true)]
-        [InlineData(false, false)]
-        public void InvalidUTF16(bool formatted, bool skipValidation)
+        [InlineData(true,
+            "{" + InvalidUtf16 + ":" + InvalidUtf16 + "}",
+            "{" + InvalidUtf16 + ":" + ValidUtf16 + "}",
+            "{" + ValidUtf16 + ":" + InvalidUtf16 + "}",
+            "{" + ValidUtf16 + ":" + ValidUtf16 + "}")]
```

Comment:
nit: Make these separate test cases rather than several expected strings in one test